bin/PullAndProcessGsAnalysisData.sh

```diff
@@ -0,0 +1,429 @@
+	#
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Processing batch ${_batch}..."
+	#
+	# Check if function previously finished successfully for this data.
+	#
+	if [[ -e "${_controlFileBaseForFunction}.finished" ]]
+	then
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${_controlFileBaseForFunction}.finished is present -> Skipping."
+		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${FUNCNAME[0]} ${_batch}. OK"
+		return
+	else
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${_controlFileBaseForFunction}.finished not present -> Continue..."
+		printf '' > "${_controlFileBaseForFunction}.started"
+	fi
+	
+	_numberOfSamplesheets=$(find "${TMP_ROOT_DIR}/tmp/${_batch}/" -maxdepth 1 -mindepth 1 -name 'CSV_UMCG_*.'"${SAMPLESHEET_EXT}" 2>/dev/null | wc -l)
+	if [[ "${_numberOfSamplesheets}" -eq 1 ]]
+	then
+		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Found: one ${TMP_ROOT_DIR}/tmp/${_batch}/CSV_UMCG_*.${SAMPLESHEET_EXT} samplesheet."
+	elif [[ "${_numberOfSamplesheets}" -gt 1 ]]
+	then
+		log4Bash 'ERROR' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "More than one CSV_UMCG_*.${SAMPLESHEET_EXT} GS samplesheet present in ${TMP_ROOT_DIR}/${_batch}/."
+		mv "${_controlFileBaseForFunction}."{started,failed}
+		return
+	elif [[ "${_numberOfSamplesheets}" -lt 1 ]]
+	then
+		log4Bash 'ERROR' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "No GS samplesheet present in ${TMP_ROOT_DIR}/${_batch}/."
+		mv "${_controlFileBaseForFunction}."{started,failed}
+		return
+	fi
+	_gsSampleSheet=$(ls -1 "${TMP_ROOT_DIR}/tmp/${_batch}/CSV_UMCG_"*".${SAMPLESHEET_EXT}")
+	
+	#
+	# Count Bam/gVCF files present on disk 
+	#
+	
+	local _countSamplesInSamplesheet
+	local _countBamFilesOnDisk
+	local _countgVcfFilesOnDisk
+
+	_countSamplesInSamplesheet=$(grep -o "${_batch}-[0-9][0-9]*" "${_gsSampleSheet}" | sort -u | wc -l)
+	_countBamFilesOnDisk=$(find "${TMP_ROOT_DIR}/tmp/${_batch}/${analysisFolder}/" -maxdepth 2 -mindepth 2 -name '*bam' | wc -l)
+	_countgVcfFilesOnDisk=$(find "${TMP_ROOT_DIR}/tmp/${_batch}/${analysisFolder}/" -maxdepth 2 -mindepth 2 -name '*.gvcf.gz' | wc -l)
+	if [[ "${_countBamFilesOnDisk}" -ne "${_countSamplesInSamplesheet}" ]]
+	then
+		log4Bash 'ERROR' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Amount bam files (${_countBamFilesOnDisk}) is not the same as the number of lines in the samplesheet ${_countSamplesInSamplesheet} (${_countSamplesInSamplesheet})."
+		mv "${_controlFileBaseForFunction}."{started,failed}
+		return
+	fi
+	if [[ "${_countgVcfFilesOnDisk}" -ne "${_countSamplesInSamplesheet}" ]]
+	then
+		log4Bash 'ERROR' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Amount gVCF files (${_countgVcfFilesOnDisk}) is not the same as the number of lines in the samplesheet ${_countSamplesInSamplesheet} (${_countSamplesInSamplesheet})."
+		mv "${_controlFileBaseForFunction}."{started,failed}
+		return
+	fi
+
+	for i in "${TMP_ROOT_DIR}/tmp/${_batch}/${analysisFolder}/"*"/"
+	do
+		cd "${i}"
+		for j in "${i}/"*".md5sum" 
```

Comment:
This relies on shell expansion of the `*`, which may not do what you want. Use `find` to make a list of md5sum files instead, then iterate over that list.

Reply:
check

Reply:
check